app/main.py

```diff
@@ -1,6 +1,14 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    stored_runs = {}
+
+    def wrapper(*args) -> Any:
+        if args in stored_runs:
+            print("Getting from cache")
+            return stored_runs[args]
+        print("Calculating new result")
+        stored_runs.update({args: func(*args)})
```

Comment:
Use `stored_runs[args] = func(*args)`, your approach introduces unnecessary complexity without any significant advantage. Also it will be more pythonic